bokehjs/src/coffee/models/tools/edit/poly_edit_tool.ts

```diff
@@ -0,0 +1,233 @@
+import {Keys} from "core/dom"
+import * as p from "core/properties"
+import {MultiLine} from "models/glyphs/multi_line"
+import {Patches} from "models/glyphs/patches"
+import {XYGlyph} from "models/glyphs/xy_glyph"
+import {GlyphRenderer} from "models/renderers/glyph_renderer"
+import {ColumnDataSource} from "models/sources/column_data_source"
+import {EditTool, EditToolView} from "./edit_tool"
+
+export interface HasXYCDS {
+  data_source: ColumnDataSource
+  glyph: XYGlyph
+}
+
+export interface HasPolyCDS {
+  data_source: ColumnDataSource
+  glyph: MultiLine | Patches
+}
+
+export interface BkEv {
+  bokeh: {
+    sx: number
+    sy: number
+  }
+  srcEvent: {
+    shiftKey?: boolean
+  }
+  keyCode: number
+  shiftKey: boolean
+}
+
+export class PolyEditToolView extends EditToolView {
+  model: PolyEditTool
+  _selected_renderer: GlyphRenderer | null
+  _basepoint: [number, number] | null
+
+  _doubletap(e: BkEv): void {
+    // Perform hit testing
+    const renderers = this._select_event(e, false, this.model.renderers);
+
+    // If we did not hit an existing line, clear node CDS
+    const vertex_renderer = this.model.vertex_renderer;
+    const point_ds = vertex_renderer.data_source;
+    const point_glyph = vertex_renderer.glyph;
+    const [pxkey, pykey] = [point_glyph.x.field, point_glyph.y.field];
```

Comment:
`field` is not a required member of a `NumberSpec`, so unless you guard against this somewhere else, this code may fail.

Reply:
Thanks I did mean to guard against that everywhere but seemingly missed this file.